macros/contexts/contextSignificantFigures.pl

```diff
@@ -118,15 +113,15 @@ sub new {
 				$digits =~ s/^0+//;                      # remove leading zeros
 			}
 		}
-		$n = length($digits) unless defined($n) && $n =~ /^inf$/;    # what remains are the significant digits
+		$n = length($digits);                            # what remains are the significant digits
```

Comment:
I would have the comment indented like the one on line 107, since it is at the same nesting depth as that, not the ones above.  But perhaps perltidy won't allow that.